File: Creational_Patterns/Singleton/Java/Eager.java
class Eager{
  private static final Eager instance = new Eager();

  private Eager(){}

  public static Eager getInstance(){
    return instance;
  }
}
 
class Main{
  public static void main(String args[]){
    Eager ob1 = Eager.getInstance();
    Eager ob2 = Eager.getInstance();
    // Eager(); get error
    System.out.println(ob1 == ob2);
  }
}

File: Creational_Patterns/Singleton/python/Eager.py
# this is the Eager way where we are only get a single instance when we call

#Note:python has the same Eager and Lazy
class Singleton:
  _instance = None

  def __init__(self):
    if Singleton._instance is None:
      Singleton._instance = self
    else:
      raise Exception("instance is already there")

  @classmethod
  def get_instance(cls):
    if cls._instance is None:
      cls._instance = Singleton()
    return cls._instance


ob1 = Singleton.get_instance()
ob2 = Singleton.get_instance()
# Singleton() # it will raise the error
print(ob1 is ob2)
